CMakeLists.txt

```diff
@@ -159,21 +161,55 @@ set(FB_SERVICE_NAME "gds_db")
 set(FB_SERVICE_PORT 3050)
 
 if (WIN32)
-    set(FB_PREFIX "c:\\\\Program Files\\\\Firebird\\\\")
+    if (MSVC)
+        set(FB_PREFIX "c:\\\\Program Files\\\\Firebird\\\\")
+    else()
+        set(FB_PREFIX ${CMAKE_INSTALL_PREFIX})
+    endif()
     set(FB_IPC_NAME "FIREBIRD")
 endif()
 
+if (MINGW)
+    set(FB_BINDIR "${FB_PREFIX}/bin")
```

Comment:
My problem with this is that it makes a windows build to use fixed instead of relocatable paths.
In the past mingw was a Windows build tool, but then seems it (together with msys) become a packaging environment too. I'm not sure just a build using mingw should follow its package conventions.

I never used cmake with Firebird, but I like it (or dislike less than the configure+make approach) and maybe we can have a cmake build with mingw compatible with the current Windows MSVC build.

Don't see this comment as an objection, as a short term solution it may be ok for me.

Does it also work using MSVC?

Reply:
> My problem with this is that it makes a windows build to use fixed instead of relocatable paths.

`FB_PREFIX` is only used on build-time. The real final installation location is defined by `FB_INSTALL_PREFIX` (at least for mingw). MinGW does not support binreloc, so we add `pathtools.cpp` and `pathtools.h` to get relocatable paths.


>In the past mingw was a Windows build tool, but then seems it (together with msys) become a packaging environment too. I'm not sure just a build using mingw should follow its package conventions.

MinGW generally follows FHS convention.. but, if you guys prefer for it to follow MSVC directory structure, im ok with that. MSys2 can always patch it downstream.

> I never used cmake with Firebird, but I like it (or dislike less than the configure+make approach) and maybe we can have a cmake build with mingw compatible with the current Windows MSVC build.
> 
> Don't see this comment as an objection, as a short term solution it may be ok for me.

Great to hear that :)

> Does it also work using MSVC?

I never tried this setup on msvc. sorry..